engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceCommandProcessor.java

```diff
@@ -30,7 +31,8 @@ public ProcessInstanceCommandProcessor(
   }
 
   @Override
-  public void processRecord(final TypedRecord<ProcessInstanceRecord> record) {
+  public void processRecord(
+      final TypedRecord<ProcessInstanceRecord> record, final SideEffects sideEffects) {
```

Comment:
Kind of funny, since we had quite a similar interface before the engine abstraction, but a level higher.

Would be happy if we can solve this with returning results instead of passing the effects as parameter.